### ISSUE TITLE
Breaking change when bumping from micronaut-openapi:4.8.6 to 4.8.7 

### ISSUE THREAD

**Issue by roar-skinderviken** [micronaut-projects/micronaut-openapi#989]
### Expected Behavior

All tests should pass as they do with micronaut-openapi:4.8.6

### Actual Behaviour

I get the following stack-trace when running tests:
```
Invalid relative name: META-INF/swagger/${api.title}-${api.version}.yml
[ERROR]   java.lang.IllegalArgumentException: Invalid relative name: META-INF/swagger/${api.title}-${api.version}.yml
[ERROR]         at jdk.compiler/com.sun.tools.javac.file.JavacFileManager.getFileForOutput(JavacFileManager.java:898)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacFiler.createResource(JavacFiler.java:541)
[ERROR]         at io.micronaut.annotation.processing.AnnotationProcessingOutputVisitor$GeneratedFileObject.getOutputObject(AnnotationProcessingOutputVisitor.java:317)
... left out for brevity
[ERROR] error: Error finalizing type visitor [io.micronaut.openapi.visitor.OpenApiApplicationVisitor@3c0d612b]: Invalid relative name: META-INF/swagger/${api.title}-${api.version}.yml
[ERROR] Note: Creating bean classes for 10 type elements
[ERROR] warning: Error:
[ERROR]   Invalid relative name: META-INF/swagger/${api.title}-${api.version}.yml
[ERROR]   java.lang.IllegalArgumentException: Invalid relative name: META-INF/swagger/${api.title}-${api.version}.yml
[ERROR]         at jdk.compiler/com.sun.tools.javac.file.JavacFileManager.getFileForOutput(JavacFileManager.java:898)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacFiler.createResource(JavacFiler.java:541)
[ERROR]         at io.micronaut.annotation.processing.AnnotationProcessingOutputVisitor$GeneratedFileObject.getOutputObject(AnnotationProcessingOutputVisitor.java:317)
```

### Steps To Reproduce

Controller method (the only one in the app)

```
@Post(
    value = "/some-path",
    consumes = [MediaType.MULTIPART_FORM_DATA],
    produces = [MediaType.APPLICATION_JSON]
)
@Operation(
    summary = "Runs controls on supplied data file/input parameters and returns a report"
)
@ApiResponses(
    ApiResponse(
        responseCode = "200",
        description = "OK",
        content = [Content(mediaType = "application/json", schema = Schema(implementation = FileReportVm::class))]
    ),
    ApiResponse(
        responseCode = "400",
        description = "Bad request",
        content = [Content(mediaType = "application/json", schema = Schema(implementation = ApiError::class))]
    ),
    ApiResponse(
        responseCode = "500",
        description = "System error",
        content = [Content(mediaType = "application/json", schema = Schema(implementation = ApiError::class))]
    )
)
@SingleResult
fun validateForm(
    someJson: String,
    file: StreamingFileUpload
): Mono<HttpResponse<FileReportVm>> {
   ... left out for brevity
```

application.yml (same stack-trace when commenting out swagger-section)
```
micronaut:
  router:
    static-resources:
      swagger:
        paths: classpath:META-INF/swagger
        mapping: /swagger/**

api:
  title: "... API"
  version: "1.0"
  description: "API for ..."
```

### Environment Information

Micronaut 3.9.0
micronaut-openapi:4.8.7
swagger-annotations:2.2.9
Kotlin 1.8.10
Maven 3.8.7
Ubuntu 22.04
JDK 17

Note: POM does not have micronaut-parent as parent-POM.

### Example Application

https://github.com/roar-skinderviken/micronaut-openapi-issue-989

### Version

micronaut-openapi:4.8.7

**Comment by altro3** [comment]
@roar-skinderviken Ok, I'll check it on the weekend. But could you give me more information about your problem: for example, OpenApiDefinition annotation values. Of course the best variant - give me example app to reproduce your problem

**Comment by roar-skinderviken** [comment]
Thanks @altro3. I'll put together an app during the weekend.

**Comment by altro3** [comment]
@roar-skinderviken Yes, I really need an example from you. I tried to reproduce the problem, but I couldn't :-(

**Comment by roar-skinderviken** [comment]
@altro3 https://github.com/roar-skinderviken/micronaut-openapi-issue-989

**Comment by altro3** [comment]
@roar-skinderviken Thanks! Interesting bug. I dont know why, but this problem reproduced only with kotlin. I did fast fix  here, but need to solve main problem. See this: https://github.com/micronaut-projects/micronaut-core/issues/9148

**Comment by roar-skinderviken** [comment]
Issue is resolved by @altro3 through MN:3.9.1 and may be closed.
